### ISSUE TITLE
Parallel recording

### ISSUE THREAD

**Issue by kmaki565** [sskodje/ScreenRecorderLib#92]
Thanks for this useful library.
When I try to capture a snapshot while I'm recording video in the same process, snapshot fails with "The parameter is incorrect.".
I can capture from a separate process.
Is this a limitation of Media Foundation API, or a limitation of the library?

**Comment by sskodje** [comment]
The snapshot is only coded to work on its own like a screenshot feature, not on a running recording.

**Comment by kmaki565** [comment]
Maybe you misunderstood my question... 
What I want to achieve is to "peek" what is being captured in an ongoing video recording. I assume the [Media Foundation's thumbnail feature](https://docs.microsoft.com/en-us/windows/win32/medfound/videothumbnail-sample) does not work on a video which is not finalized yet, so I plan to capture snapshots while I capture a video, preferably from a single app.
This video [ReproVideo.zip](https://github.com/sskodje/ScreenRecorderLib/files/5323233/ReproVideo.zip) describes the issue I faced. I added a new button in the TestApp that creates one more `MainWindow`. You can see taking snapshot fails while video is recorded. 
I found the same problem NOT occurring if I target different displays for video and snapshot. So I am now assuming this is a limitation of the Media Foundation API.
Please let me know if you have a different view. Thanks.

**Comment by kmaki565** [comment]
I found the answer by myself... 
It failed with IDXGIOutput1::DuplicateOutput and [the doc](https://docs.microsoft.com/ja-jp/windows/win32/api/dxgi1_2/nf-dxgi1_2-idxgioutput1-duplicateoutput?redirectedfrom=MSDN) says _A process can have only one desktop duplication interface on a single desktop output_. So this is as-designed behavior. 
Also, looking into the library code, I came up with the idea that I could capture snapshots using frames which I write to a video in some interval (within `internal_recorder::RenderFrame`). I think I can go with this idea.
Thank you for your time!
